urllib3/util/selectors.py

```diff
@@ -0,0 +1,544 @@
+# Backport of selectors.py from Python 3.5+ to support Python < 3.4
```

Comment:
For the moment I'd rather it lives here, I think. Given that this is actually a fairly customised backport that we are trying to enforce code coverage on, I think we need to consider it part of our product rather than something that we just trust to work.